### ISSUE TITLE
dsc_base_provider error on resource parameters which are Booleans

### ISSUE THREAD

**Issue by bwilcox** [puppetlabs/ruby-pwsh#84]
## Describe the Bug
I'm using the dsc_userrightsassignment resource from securitypolicydsc.  I added the parameter 'dsc_force' to my resource and set it to 'true'  (Boolean, not a string).  When attempting to apply the resource I get an error message:

```
Error: dsc_userrightsassignment[{:name=>"User network access", :dsc_policy=>"Access_this_computer_from_the_network"}]: Updating: Failed after 0.010350 seconds: undefined method `empty?' for true:TrueClass
Error: /Stage[main]/Main/Dsc_userrightsassignment[User network access]: Could not evaluate: Execution encountered an error
```

## Expected Behavior
`dsc_force` is a valid parameter and should be set as a Boolean.  I expect the resource to function normally with this parameter set.

## Steps to Reproduce
This is the resource I'm testing with:

```
  $users = [
    'Builtin\Administrators',
    'NT Authority\Authenticated Users'
  ]

  dsc_userrightsassignment{'User network access':
    dsc_identity => $users,
    dsc_policy   => 'Access_this_computer_from_the_network',
    dsc_force    => true,
  }
```

## Environment
 - Version pwshlib 0.6.3, securitypolicydsc 2.10.0-0-1
 - Platform Windows 2019

## Additional Context
I found this while attempting to figure out why 21 of 23 resources using `dsc_userrightsassignment` were not behaving in an idempotent fashion.

**Comment by michaeltlombardi** [comment]
Thanks for the repro! We'll take a look.

**Comment by michaeltlombardi** [comment]
Resolved in #90
